pkg/osbuild/fips.go

```diff
@@ -0,0 +1,56 @@
+package osbuild
+
+import (
+	"github.com/osbuild/images/internal/common"
+	"github.com/osbuild/images/internal/fsnode"
+	"github.com/osbuild/images/pkg/disk"
+	"os"
+)
+
+func GenFIPSKernelOptions(pt *disk.PartitionTable) []string {
+	cmdline := make([]string, 0)
+	cmdline = append(cmdline, "fips=1")
+
+	genOptions := func(e disk.Entity, path []disk.Entity) error {
+		switch ent := e.(type) {
+		case *disk.Filesystem:
+			if ent.Mountpoint == "/boot" {
+				if label := ent.Label; label != "" {
+					karg := "boot=LABEL=" + label
+					cmdline = append(cmdline, karg)
+				} else if uuid := ent.UUID; uuid != "" {
+					karg := "boot=UID=" + uuid
+					cmdline = append(cmdline, karg)
+				}
+			}
+		}
+		return nil
+	}
+
+	_ = pt.ForEachEntity(genOptions)
```

Comment:
Since we only need to do something for `/boot`, you can use:
```go
	if bootMnt := pt.FindMountable("/boot"); bootMnt != nil {
		boot := bootMnt.GetFSSpec()
		if label := boot.Label; label != "" {
			karg := "boot=LABEL=" + label
			cmdline = append(cmdline, karg)
		} else if uuid := boot.UUID; uuid != "" {
			karg := "boot=UID=" + uuid
			cmdline = append(cmdline, karg)
		}
	}

```
which I find more readable.

Reply:
Done